frame/staking/src/lib.rs

```diff
@@ -783,6 +783,32 @@ decl_event!(
 	}
 );
 
+decl_error! {
+	/// Error for the stacking module.
+	pub enum Error {
+		/// Not a controller account.
+		NotController,
+		/// Not a stash account.
+		NotStash,
+		/// Stash is already bonded.
+		AlreadyBonded,
+		/// Controller is already paired.
+		AlreadyPaired,
+		/// Should be the root origin or the `T::SlashCancelOrigin`.
+		BadOrigin,
```

Comment:
```suggestion
		InvalidOrigin,
```

Let's use the term "InvalidOrigin" rather than "BadOrigin" :)